cosmo_tester/test_suites/snapshots/__init__.py

```diff
@@ -24,8 +33,204 @@
 
 
 HELLO_WORLD_URL = 'https://github.com/cloudify-cosmo/cloudify-hello-world-example/archive/4.0.zip'  # noqa
+# We need this because 3.4 (and 4.0!) snapshots don't handle agent_config,
+# but 3.4 example blueprints use it instead of cloudify_agent
+OLD_WORLD_URL = 'https://github.com/cloudify-cosmo/cloudify-hello-world-example/archive/3.3.1.zip' # noqa
+BASE_ID = 'helloworld'
+BLUEPRINT_ID = '{base}_bp'.format(base=BASE_ID)
+DEPLOYMENT_ID = '{base}_dep'.format(base=BASE_ID)
+NOINSTALL_BLUEPRINT_ID = '{base}_noinstall_bp'.format(base=BASE_ID)
+NOINSTALL_DEPLOYMENT_ID = '{base}_noinstall_dep'.format(base=BASE_ID)
+SNAPSHOT_ID = 'testsnapshot'
+# This is used purely for testing that plugin restores have occurred.
+# Any plugin should work.
+TEST_PLUGIN_URL = 'http://repository.cloudifysource.org/cloudify/wagons/cloudify-openstack-plugin/2.0.1/cloudify_openstack_plugin-2.0.1-py27-none-linux_x86_64-centos-Core.wgn'  # noqa
+BASE_PLUGIN_PATH = '/opt/mgmtworker/env/plugins/{tenant}/'
+INSTALLED_PLUGIN_PATH = BASE_PLUGIN_PATH + '{name}-{version}'
+FROM_SOURCE_PLUGIN_PATH = BASE_PLUGIN_PATH + '{deployment}-{plugin}'
+DEPLOYMENT_ENVIRONMENT_PATH = (
+    '/opt/mgmtworker/work/deployments/{tenant}/{name}'
+)
+
+
+def upgrade_agents(cfy, manager, logger, tenants=('default_tenant',)):
+    manager.use()
+    for tenant in tenants:
+        logger.info('Upgrading agents for {tenant}'.format(tenant=tenant))
+        cfy.agents.install(['-t', tenant])
+
+
+def remove_and_check_deployments(hello_vms, manager, logger,
+                                 tenants=('default_tenant',),
+                                 with_prefixes=False):
+    for tenant in tenants:
+        logger.info(
+            'Uninstalling hello world deployments from manager for '
+            '{tenant}'.format(
+                tenant=tenant,
+            )
+        )
+        logger.info('Found deployments: {deployments}'.format(
+            deployments=', '.join(get_deployments_list(manager, tenant)),
+        ))
+        with set_client_tenant(manager, tenant):
+            if with_prefixes:
+                deployment_id = tenant + DEPLOYMENT_ID
+            else:
+                deployment_id = DEPLOYMENT_ID
+            execution = manager.client.executions.start(
+                deployment_id,
+                'uninstall',
+            )
+
+        logger.info('Waiting for uninstall to finish')
+        wait_for_execution(
+            manager,
+            execution,
+            logger,
+            tenant,
+        )
+        logger.info('Uninstalled deployments for {tenant}'.format(
+            tenant=tenant,
+        ))
+
+    assert_hello_worlds(hello_vms, installed=False, logger=logger)
+
+
+def delete_manager(manager, logger):
+    logger.info('Deleting {version} manager..'.format(
+        version=manager.branch_name))
+    manager.delete()
+
+
+def create_noinstall_helloworld(manager, logger, tenant=None):
```

Comment:
Let's maybe call it something different. It's a really confusing name, especially considering what it actually does.